cirq-core/cirq/optimizers/two_qubit_to_sqrt_iswap.py

```diff
@@ -0,0 +1,340 @@
+"""Utility methods for decomposing two-qubit unitaries into sqrt-iSWAP gates.
+
+References:
+    Towards ultra-high fidelity quantum operations: SQiSW gate as a native
+    two-qubit gate
+    https://arxiv.org/abs/2105.06074
+"""
+
+from typing import Optional, Sequence, Tuple, TYPE_CHECKING
+
+import numpy as np
+
+from cirq import ops, linalg, protocols
+from cirq.optimizers import two_qubit_decompositions
+
+if TYPE_CHECKING:
+    import cirq
+
+
+def two_qubit_matrix_to_sqrt_iswap_operations(
+    q0: 'cirq.Qid',
+    q1: 'cirq.Qid',
+    mat: np.ndarray,
+    *,
+    required_sqrt_iswap_count: Optional[int] = None,
+    atol: float = 1e-8,
+    check_preconditions: bool = True,
+    clean_operations: bool = True,
+) -> Sequence['cirq.Operation']:
+    """Decomposes a two-qubit operation into Z/XY/sqrt-iSWAP gates.
+
```

Comment:
It might be good to give a little bit more detail about the implementation details for those too lazy or tired to read the paper.  For instance, mention that this decomposes into the unitary's KAK decomposition.

Reply:
Done.